### PR TITLE
Use fbp-graph for Graph and Journal

### LINKED ISSUE #517: Split Graph code to dedicated library
Wanted for use [in the-graph](https://github.com/flowhub/the-graph/issues/314) (which is not NoFlo-specific), and possibly [for MicroFlo](https://github.com/microflo/microflo/issues/42)

### PR DESCRIPTION
**Pull request by bergie**
fixes #517

### CODE REVIEW

**Comment by bergie** [comment]
The versioneye fail is probably because they didn't fetch the new library yet. NPM shows license correctly: https://www.npmjs.com/package/fbp-graph

File: src/lib/loader/NodeJs.coffee [hunk 1]
```diff
@@ -170,7 +170,7 @@ exports.getSource = (loader, name, callback) ->
     nameParts[0] = ''
 
   if loader.isGraph component
-    nofloGraph.loadFile component, (err, graph) ->
+    fbpGraph.loadFile component, (err, graph) ->
```
Comment (jonnor) [inline src/lib/loader/NodeJs.coffee]: Other places this is `fbpGraph.graph.loadFile`?

File: src/lib/ComponentLoader.coffee [hunk 2]
```diff
@@ -112,7 +112,7 @@ class ComponentLoader extends EventEmitter
 
   isGraph: (cPath) ->
     # Live graph instance
-    return true if typeof cPath is 'object' and cPath instanceof nofloGraph.Graph
+    return true if typeof cPath is 'object' and cPath instanceof fbpGraph.Graph
```
Comment (jonnor) [inline src/lib/ComponentLoader.coffee]: Hm, will this break compatibility? If people have an existing nofloGraph.Graph that they are passing in?

File: src/lib/loader/ComponentIo.coffee [hunk 3]
```diff
@@ -138,7 +138,7 @@ exports.getSource = (loader, name, callback) ->
     nameParts[0] = ''
 
   if loader.isGraph component
-    nofloGraph.loadFile component, (err, graph) ->
+    fbpGraph.loadFile component, (err, graph) ->
```
Comment (jonnor) [inline src/lib/loader/ComponentIo.coffee]: Same here, isn't. Might be best to make the imports be `fbpGraph = require('fbp-graph').graph` ?

File: src/lib/ComponentLoader.coffee (hunk 2 above)
Comment (bergie) [inline src/lib/ComponentLoader.coffee]: it should work given that `noflo.Graph` is now just an alias of `fbp-graph.Graph`

File: src/lib/loader/NodeJs.coffee (hunk 1 above)
Comment (bergie) [inline src/lib/loader/NodeJs.coffee]: Fixed in a93cff8

File: src/lib/ComponentLoader.coffee (hunk 2 above)
Comment (jonnor) [inline src/lib/ComponentLoader.coffee]: ... assuming that the code uses the exact same NoFlo version? But I guess that may have been a requirement before also.

Comment (bergie) [inline src/lib/ComponentLoader.coffee]: Yep. Might be good to change that to a more duck-typed check, but that can be done separately from this.

**Comment by jonnor** [comment]
Was commented out in d1a55d3d778ca1afb484fda1ce8727033efec721

**Comment by jonnor** [comment]
The test was introduced in 311d724f2907db15a0df92f4c40f077f2f45675f and probably its expectations are not right for current behavior

**Comment by bergie** [comment]
@jonnor added tests in 1d29626

**Pull request merged**